test/__tests__/data/ga4.json

```diff
@@ -11799,5 +11807,422 @@
       "version": "1",
       "endpoint": "https://www.google-analytics.com/mp/collect"
     }
-  }
-]
+  },
+  {
```

Comment:
@mihir-4116 Write a unit test case for the util function. This test suite is more functional one.

Reply:
This is not readable way to know whether all cases got handled for a unit of function.

Reply:
added tests for the util function